### PR TITLE
Add QAOA example

### PR DESCRIPTION
**Pull request by kevinsung**
At first I thought I would want to use a Display, but the measurements here are not complicated enough to warrant that.

Another note: at small system sizes (even up to 16 qubits) this example pretty much always finds the optimal answer due to the fact that we sample so many bitstrings. I set n=6 so that the example circuit in the docstring would fit within 80 lines.

### CODE REVIEW

File: examples/qaoa.py [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+                        for bitstring in itertools.product(range(2), repeat=n))
+
+    # Print the results
+    print('The largest cut value found was {}.'.format(largest_cut_value_found))
+    print('The largest possible cut has size {}.'.format(max_cut_value))
+    print('The approximation ratio achieved is {}.'.format(
+        largest_cut_value_found / max_cut_value))
+
+
+def Rzz(rads):
```
Comment (dabacon) [inline examples/qaoa.py]: nit: style would say make this rzz

File: examples/qaoa.py [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+Optimizing objective function ...
+The largest cut value found was 7.
+The largest possible cut has size 7.
+The approximation ratio achieved is 1.0.
+"""
+
+import itertools
+
+import networkx
+import numpy as np
```
Comment (dabacon) [inline examples/qaoa.py]: examples/qaoa.py:54:0: C0411: third party import "import numpy as np" should be placed before "import networkx" (wrong-import-order)

Comment (kevinsung) [reply examples/qaoa.py]: Done

File: examples/qaoa.py (hunk 1 above)
Comment (kevinsung) [reply examples/qaoa.py]: We have cirq.Rz though

File: examples/__init__.py [hunk 3]
```diff
@@ -1,4 +1,4 @@
-# Copyright 2018 The Cirq Developers
+# Copyright 2018 The Cirq Developersdef test_example_runs_basic_arithmetic():
```
Comment (balopat) [inline examples/__init__.py]:
```suggestion
# Copyright 2018 The Cirq Developers```

File: examples/qaoa.py (hunk 1 above)
Comment (dabacon) [reply examples/qaoa.py]: cirq.Rz is a class.  Here you have a method.

**Comment by kevinsung** [comment]
Actually, cirq.Rz is a method:
```
def Rz(rads: Union[float, sympy.Basic]) -> ZPowGate:
    """Returns a gate with the matrix e^{-i Z rads / 2}."""
    pi = sympy.pi if protocols.is_parameterized(rads) else np.pi
    return ZPowGate(exponent=rads / pi, global_shift=-0.5)
```

**Comment by dabacon** [comment]
Method name should not be `Rx` in Cirq.

**Pull request merged**